### PR TITLE
Add event repository

### PR DESCRIPTION
**Pull request by kuotinyen**

### CODE REVIEW

**Comment by kuotinyen** [comment]
It should be on my branch to pull-request, but since recently I have no time to redo it, so please forgive me hahaha.

File: IOS@Taipei/IOS@Taipei/AppDelegate.swift [hunk 1]
```diff
@@ -19,7 +19,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        FirebaseApp.configure()
+        FirebaseApp.configure()
```
Comment (andrew54068) [inline IOS@Taipei/IOS@Taipei/AppDelegate.swift]: please don't add additional change if it doesn't mean anything.

File: IOS@Taipei/Podfile.lock [hunk 2]
```diff
@@ -123,4 +123,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 5a71a87ef7d2a8c516ad4eeea58ab00e5e88ce05
 
-COCOAPODS: 1.8.3
+COCOAPODS: 1.8.4
```
Comment (andrew54068) [inline IOS@Taipei/Podfile.lock]: please don't make this change if we are not yet agree with using this version.

File: IOS@Taipei/IOS@Taipei/Model/GroupEvent/GroupEvent.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        case maybeCount = "maybe_count"
+        case noreplyCount = "noreply_count"
+        case cover
+        case description
+        case name, place
+        case startTime = "start_time"
+        case endTime = "end_time"
+        case updatedTime = "updated_time"
+        case type, id
+    }
```
Comment (andrew54068) [inline IOS@Taipei/IOS@Taipei/Model/GroupEvent/GroupEvent.swift]: if using JSONDecoder().keyDecodingStrategy = .convertFromSnakeCase
this is not necessary.

File: IOS@Taipei/IOS@Taipei/Repository/GroupEventRepository.swift [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+ final class GroupEventRepository {
+    static let shared = GroupEventRepository()
+    private init() {}
+
+     private var accessToken: String?
+
+     func setup(with accessToken: String) {
+        self.accessToken = accessToken
+    }
```
Comment (andrew54068) [inline IOS@Taipei/IOS@Taipei/Repository/GroupEventRepository.swift]: what this for?

File: IOS@Taipei/IOS@Taipei/Repository/GroupEventRepository.swift [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    case unknown(Error?)
+    case noResponseData
+    case noDataModel
+    case noFBAccessToken
+}
+
+ final class GroupEventRepository {
+    static let shared = GroupEventRepository()
+    private init() {}
+
```
Comment (andrew54068) [inline IOS@Taipei/IOS@Taipei/Repository/GroupEventRepository.swift]: Is there any indent misplace?

Comment (kuotinyen) [reply IOS@Taipei/IOS@Taipei/Repository/GroupEventRepository.swift:22]: what do you mean?

File: IOS@Taipei/IOS@Taipei/AppDelegate.swift (hunk 1 above)
Comment (kuotinyen) [reply IOS@Taipei/IOS@Taipei/AppDelegate.swift]: ok

File: IOS@Taipei/Podfile.lock (hunk 2 above)
Comment (kuotinyen) [reply IOS@Taipei/Podfile.lock]: ok

File: IOS@Taipei/IOS@Taipei/Model/GroupEvent/GroupEvent.swift (hunk 3 above)
Comment (kuotinyen) [reply IOS@Taipei/IOS@Taipei/Model/GroupEvent/GroupEvent.swift]: ok

File: IOS@Taipei/IOS@Taipei/Repository/GroupEventRepository.swift (hunk 4 above)
Comment (kuotinyen) [reply IOS@Taipei/IOS@Taipei/Repository/GroupEventRepository.swift:27]: what do you mean?

Comment (andrew54068) [reply IOS@Taipei/IOS@Taipei/Repository/GroupEventRepository.swift]: how should we use setup function?

File: IOS@Taipei/IOS@Taipei/Repository/GroupEventRepository.swift (hunk 5 above)
Comment (andrew54068) [reply IOS@Taipei/IOS@Taipei/Repository/GroupEventRepository.swift]: select this area and press control + i and see what happened.

File: IOS@Taipei/IOS@Taipei/Model/GroupEvent/GroupEvent.swift [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        case maybeCount
+        case noreplyCount
+        case cover
+        case description
+        case name, place
+        case startTime
+        case endTime
+        case updatedTime
+        case type, id
+    }
```
Comment (andrew54068) [inline IOS@Taipei/IOS@Taipei/Model/GroupEvent/GroupEvent.swift]: this codingKeys is redundant if each key matches all variable.

**Pull request closed without merging**